install.sh

```diff
@@ -77,88 +76,172 @@ ros_git_get() {
+				echo "Several dependencies no longer exist in ROS Kinetic, so in order to install it,"
+				echo "ROS Indigo, the Sub8 repository at an earlier date,  and all of the old Sub8"
+				echo "dependencies will need to be downloaded and installed."
+				echo -n "Do you still wish to proceed? [y/N] " && read RESPONSE
+				echo ""
+				if ([ "$RESPONSE" = "Y" ] || [ "$RESPONSE" = "y" ]); then
+					if (cat $BASHRC_FILE | grep --quiet "source /opt/ros/kinetic/setup.bash") && \
+					   !(cat $BASHRC_FILE | grep --quiet "#source /opt/ros/kinetic/setup.bash"); then
+						instwarn "Terminating installation due to conflicting ROS versions"
+						instwarn "NaviGator requires ROS Indigo, but ROS Kinetic is being sourced"
+						instwarn "To fix this issue, comment out this line in $BASHRC_FILE: source /opt/ros/kinetic/setup.bash"
+						exit 1
+					else
+						INSTALL_NAV=true
+						SELECTED=true
+					fi
+				fi
 			;;
-		-n) INSTALL_NAV=true
-			shift 1
+			"")
+				echo "You must select one of the projects by entering it's number on the list"
+				echo ""
 			;;
-		-?) instwarn "Option $1 is not implemented"
-			exit 1
+			*)
+				echo "$RESPONSE is not a valid selection"
+				echo ""
 			;;
-	esac
-done
+		esac
+	done
 
-if !($INSTALL_ALL || $INSTALL_SUB || $INSTALL_NAV); then
-	echo "A MIL project must be selected for install"
-	echo "Run ./install.sh -h for more information"
-	exit 1
-elif ($INSTALL_ALL); then
-	INSTALL_SUB=true
-	INSTALL_NAV=true
-elif ($INSTALL_NAV); then
-
-	# Navigator currently depends on the Sub8 repository
-	# This may change soon, but this will install Sub8 for now
-	INSTALL_SUB=true
+	# If the user chooses to install the BlueView SDK, retrieve the password from them
+	echo "The BlueView SDK used to interface with the Telodyne imaging sonar is encrypted"
+	echo "in order to protect the intellectual property of BlueView. If you will be doing"
+	echo "work with the imaging sonar on your machine, it is recommended that you install"
+	echo "this now. If not, you probably do not need to."
+	echo -n "Do you wish to install the SDK? [y/N] " && read RESPONSE
+	echo ""
+	if ([ "$RESPONSE" = "Y" ] || [ "$RESPONSE" = "y" ]); then
+		echo "The SDK is encrypted with a password. You need to obtain this password from one"
+		echo "of the senior members of MIL."
+		echo -n "Encryption password: " && read -s PASSWORD
+		echo ""
+	fi
+fi
+
+if ($INSTALL_SUB); then
+	REQUIRED_OS_CODENAME="xenial"
+	REQUIRED_OS_VERSION="16.04"
+	ROS_VERSION="kinetic"
+else
+	REQUIRED_OS_CODENAME="trusty"
+	REQUIRED_OS_VERSION="14.04"
+	ROS_VERSION="indigo"
 fi
 
-# The paths to the aliases configuration files
-BASHRC_FILE=~/.bashrc
-ALIASES_FILE=~/.mil_aliases
 
 #==================#
 # Pre-Flight Check #
 #==================#
 
+instlog "Acquiring root privileges"
+
+# Gets the user to enter their password here instead of in the middle of the pre-flight check
+# Purely here for aesthetics and to satisfy OCD
+sudo true
```

Comment:
The user check is really just there to make sure you're installing as a user that is not root which is just a bad idea and probably wouldn't work.